src/utils/values.c

```diff
@@ -840,3 +840,77 @@ sr_val_to_buff(const sr_val_t *value, char buffer[], size_t size)
 
     return len;
 }
+
+API int
+sr_tree_to_val(struct lyd_node *data, const char *path, sr_val_t **value)
+{
+    sr_error_info_t *err_info = NULL;
+    struct ly_set *set = NULL;
+
+    if(!data || !path || !value)
+        return SR_ERR_INVAL_ARG;
+
+    *value = NULL;
+
+    set = lyd_find_path(data, path);
+
+    if (!set || !set->number) {
```

Comment:
`!set` should probably be handled specially, look [here](https://github.com/sysrepo/sysrepo/blob/master/src/common.c#L3135), for example.

Reply:
Hi Michal,

Actually empty set can be possible if the requested node's module is not present in data tree.
I came across such scenarios in my testing. 

Any way node is not found, so returning NOT_FOUND error. Does it cause any unexpected? Please help me to understand.

Thanks & Regards,
Sriram.K

Reply:
Fixed as per your comment